Lib/imaplib.py

```diff
@@ -1279,6 +1385,236 @@ def print_log(self):
                 n -= 1
 
 
+class _Idler:
+    # Iterable context manager: start IDLE & produce untagged responses
+    #
+    # This iterator produces (type, datum) tuples.  They slightly differ
+    # from the tuples returned by IMAP4.response():  The second item in the
+    # tuple is a single datum, rather than a list of them, because only one
+    # untagged response is produced at a time.
+
+    def __init__(self, imap, dur=None):
+        if 'IDLE' not in imap.capabilities:
+            raise imap.error("Server does not support IDLE")
+        self._dur = dur
+        self._imap = imap
+        self._tag = None
+        self._sock_timeout = None
+        self._old_state = None
+
+    def __enter__(self):
+        imap = self._imap
+        assert not (imap._idle_responses or imap._idle_capture)
+
+        if __debug__ and imap.debug >= 4:
+            imap._mesg('idle start'
+                       + ('' if self._dur is None else f' dur={self._dur}'))
+
+        try:
+            # Start capturing untagged responses before sending IDLE,
+            # so we can deliver via iteration any that arrive while
+            # the IDLE command continuation request is still pending.
+            imap._idle_capture = True
+
+            self._tag = imap._command('IDLE')
+            # Process responses until the server requests continuation
+            while resp := imap._get_response():  # Returns None on continuation
+                if imap.tagged_commands[self._tag]:
```

Comment:
As a reviewer, I don't know exactly what this does. Either a comment explaining the case, or something more clear in the case rather than just truthiness.